File: Java/bogosort/TheAlgorithms/TheAlgorithms.java
import java.util.Random;

public class TheAlgorithms {

    private static final Random RANDOM = new Random();

    private static <T extends Comparable<T>> boolean isSorted(T[] array) {
        for (int i = 0; i < array.length - 1; i++) {
            if (array[i + 1].compareTo(array[i]) < 0) { // Comparação direta usando compareTo
                return false;
            }
        }
        return true;
    }

    // Randomly shuffles the array
    private static <T> void nextPermutation(T[] array) {
        int length = array.length;

        for (int i = 0; i < array.length; i++) {
            int randomIndex = i + RANDOM.nextInt(length - i);
            swap(array, randomIndex, i);
        }
    }

    // Swap two elements in the array
    private static <T> void swap(T[] array, int i, int j) {
        T temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }

    public <T extends Comparable<T>> T[] sort(T[] array) {
        while (!isSorted(array)) {
            nextPermutation(array);
        }
        return array;
    }

    public static void main(String[] args) {
        // Gerando 1000 valores aleatórios
        Integer[] integers = new Integer[1000];
        for (int i = 0; i < integers.length; i++) {
            integers[i] = RANDOM.nextInt(1000); // Números entre 0 e 999
        }

        TheAlgorithms bogoSort = new TheAlgorithms();

        // Ordenando o array (sem print)
        bogoSort.sort(integers);

        // O array já estará ordenado, mas sem saída de print
    }
}

File: C/bogosort/thealgorithms/thealgorithms.c
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>

bool check_sorted(int *a, int n)
{
    while (--n >= 1)
    {
        if (a[n] < a[n - 1])
            return false;
    }
    return true;
}

void shuffle(int *a, int n)
{
    int i, t, r;
    for (i = 0; i < n; i++)
    {
        t = a[i];
        r = rand() % n;
        a[i] = a[r];
        a[r] = t;
    }
}

void sort(int *a, int n)
{
    srand(time(NULL));
    while (!check_sorted(a, n)) shuffle(a, n);
}

int main()
{
    int n = 10;
    int numbers[n];

    for (int i = 0; i < n; i++)
    {
        numbers[i] = (rand() % 10000) + 1;  // Gera valores aleatórios entre 0 e 9999
    }

    sort(numbers, n);

    return 0;
}